### PR TITLE
Fix broken Scaladoc external references with JDK11

### PR DESCRIPTION
**Pull request by steven-barnes**
Constructs URL for Java API references based on java.version; can override using -jdk-api-doc-base argument. partially addresses scala/bug#11839

only supports mappings to the `java.base` module — supporting other modules remains as future work

### CODE REVIEW

**Comment by SethTisue** [comment]
tests passing — can you rebase/squash? once it's a single commit, the "combined" error will go away (the "combined" check means that every commit must be green, not just the last one)

**Comment by steven-barnes** [comment]
My branch has merges from 2.13.x. Is it safe to rebase/squash them?

**Comment by SethTisue** [comment]
yes. you should find that when you rebase, the merge commits will go away.

**Comment by steven-barnes** [comment]
Can we get moving with the code review? I will be happy to rewrite the code as needed.

File: project/ScalaOptionParser.scala [hunk 1]
```diff
@@ -124,7 +124,7 @@ object ScalaOptionParser {
   private def scalaDocBooleanSettingNames = List("-implicits", "-implicits-debug", "-implicits-show-all", "-implicits-sound-shadowing", "-implicits-hide", "-author", "-diagrams", "-diagrams-debug", "-raw-output", "-no-prefixes", "-no-link-warnings", "-expand-all-types", "-groups")
   private def scalaDocIntSettingNames = List("-diagrams-max-classes", "-diagrams-max-implicits", "-diagrams-dot-timeout", "-diagrams-dot-restart")
   private def scalaDocChoiceSettingNames = Map("-doc-format" -> List("html"))
-  private def scaladocStringSettingNames = List("-doc-title", "-doc-version", "-doc-footer", "-doc-no-compile", "-doc-source-url", "-doc-generator", "-skip-packages")
+  private def scaladocStringSettingNames = List("-doc-title", "-doc-version", "-doc-footer", "-doc-no-compile", "-doc-source-url", "-doc-generator", "-skip-packages", "jdk-api-doc-base")
```
Comment (SethTisue) [inline project/ScalaOptionParser.scala]: the other strings begin with a hyphen but the new one doesn't, was that an oversight?

Comment (SethTisue) [inline project/ScalaOptionParser.scala]: the other strings begin with a hyphen but the new one doesn't, was that an oversight?

**Review by SethTisue — COMMENTED** [review]
I expect @retronym will attempt to review this in time for the 2.13.2 release, but he might be backed up at the moment.

**Comment by SethTisue** [comment]
@ashawley does this look plausible to you?

File: project/ScalaOptionParser.scala (hunk 1 above)
Comment (steven-barnes) [reply project/ScalaOptionParser.scala]: yes, I'll change that

**Comment by SethTisue** [comment]
(retargeted for 2.13.3, but that doesn't mean that it _can't_ still make 2.13.2)

**Comment by SethTisue** [comment]
@lrytz wdyt about merging this for 2.13.2?

**Review by retronym — APPROVED** [review]
I think this is a great first step. I got a bit sidetracked when reviewing considering making things more general or performant, but we can refine this in a later release.

**Review by lrytz — APPROVED** [review]

**Comment by ashawley** [comment]
I wasn't able to look at this closely, but I recall it only supports mappings to the java.base module.   Merging this as a first step is fine, but it doesn't fix scala/bug#11839 entirely, so it should stay open.

**Comment by SethTisue** [comment]
I updated the PR description to include the `java.base` caveat

**Comment by SethTisue** [comment]
thank you @steven-barnes!

**Comment by ashawley** [comment]
This change broke Scaladoc on Windows on JDK11, see https://github.com/scala/bug/issues/11955

**Comment by steven-barnes** [comment]
We have discussed a possible solution [here](https://github.com/scala/bug/issues/11839#issuecomment-619247437), however I do not have a Windows machine to test.

**Comment by ashawley** [comment]
The discussion should continue at https://github.com/scala/bug/issues/11955 and maybe someone could assign you the ticket.

**Pull request merged**